File: uLab/Assets/Scripts/UI/BagWindow.cs


using Locke;
using Locke.ui;
using UnityEngine;
using UnityEngine.UI;

public class BagWindow : IWindow
{
	private static string[] buttonNames = new string[5] { "bag", "role", "shop", "tip", "close normals" };

	protected override void OnInit()
	{
		for (int i = 0; i < 5; i++)
		{
			Button button = this.FindWidget<Button>("buttons/button" + (i + 1));
			button.transform.GetComponentInChildren<Text>().text = buttonNames[i];
			UIEventListener.Get(button.gameObject).onClick = OnButtonClick;
		}

		Button back = FindWidget<Button>("buttons/back");
		back.onClick.AddListener(OnClick);
	}

	protected override void OnExit(IContext context)
	{
	}

	protected override void OnPause(IContext context)
	{
	}

	protected override void OnResume(IContext context)
	{
	}

	private void OnButtonClick(GameObject go)
	{
		int index = int.Parse(go.name.Substring(go.name.Length - 1));

		if (index == 1)
		{
			UIManager.Instance.OpenWindow(GameUI.bag);
		}
		else if (index == 2)
		{
			UIManager.Instance.OpenWindow(GameUI.role);
		}
		else if (index == 3)
		{
			UIManager.Instance.OpenWindow(GameUI.shop);
		}
		else if (index == 4)
		{
			UIManager.Instance.OpenWindow(GameUI.tip);
		}
		else if (index == 5)
		{
			UIManager.Instance.BackToMainWindow();
		}
	}

	private void OnClick()
	{
		UIManager.Instance.CloseWindow(this);
	}

}

File: uLab/Assets/Lua/UI/BagWindow.lua

local gameObject
local transform

local BagWindow = class("BagWindow",UIBase)
--local this = BagWindow

local buttonNames = { "bag", "role", "shop", "tip", "main"}


function BagWindow:OnInit(obj)
	gameObject = obj
	transform = obj.transform

	for i = 1, 5 do
		local btn = Util.FindGameObject(gameObject, 'buttons/button'..tostring(i))
		local label = Util.FindComponent(btn, 'Text', "Text")
		label.text = buttonNames[i]
		UIEventListener.SetOnClick(btn, self.onBtnClick, self)
	end

	local back = Util.FindGameObject(gameObject, "buttons/back")
	UIEventListener.SetOnClick(back, self.onBackClick, self)

end

function BagWindow:OnEnter()
	
end

function BagWindow:OnExit()
	
end

function BagWindow:OnPause()
	
end

function BagWindow:OnResume()
	
end

function BagWindow:onBtnClick(go)
	local index = tonumber(string.sub(go.name, string.len(go.name)))
	
	if index == 1 then
		gUIManager:OpenWindow(GameUI.bag)
	elseif index == 2 then
		gUIManager:OpenWindow(GameUI.role)
	elseif index == 3 then
		gUIManager:OpenWindow(GameUI.shop)
	elseif index == 4 then
		gUIManager:OpenWindow(GameUI.tip)
	elseif index == 5 then
		gUIManager:BackToMainWindow()
	end
end

function BagWindow:onBackClick(go)
	gUIManager:CloseWindow(GameUI.bag)
end

return BagWindow
